Makefile: add libgeos_c.so as a strict dependency

otan opened the pull request:
I got a build failure recently that had libgeos.so but not libgeos_c.so.
That was weird -- in any case, this is the fix...

----

We only did libgeos.so, which does not contain libgeos_c.so. Both are
needed - so ensure both are added.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56831)
<!-- Reviewable:end -->

otan committed:
Makefile: add libgeos_c.so as a strict dependency

We only did libgeos.so, which does not contain libgeos_c.so. Both are
needed - so ensure both are added.

Release note: None

otan commented:
https://teamcity.cockroachdb.com/viewLog.html?buildId=2453132&buildTypeId=Cockroach_UnitTests hmm this represents what i saw. the build log is not helpful:

```

[23:05:14][Compile acceptance tests] build/builder.sh mkrelease linux -Otarget testbuild TAGS=acceptance PKG=./pkg/acceptance
[23:05:14][Compile acceptance tests] + CGO_ENABLED=1
[23:05:14][Compile acceptance tests] + make BUILDTYPE=release XGOOS=linux XGOARCH=amd64 XCMAKE_SYSTEM_NAME=Linux TARGET_TRIPLE=x86_64-unknown-linux-gnu 'LDFLAGS=-static-libgcc -static-libstdc++ -lrt' SUFFIX=-linux-2.6.32-gnu-amd64 -Otarget testbuild TAGS=acceptance PKG=./pkg/acceptance
[23:05:14][Compile acceptance tests] Running make with -j16
[23:05:14][Compile acceptance tests] GOPATH set to /go
[23:05:15][Compile acceptance tests] mkdir -p lib
[23:05:15][Compile acceptance tests] rm -f lib/lib{geos,geos_c}.so
[23:05:15][Compile acceptance tests] cp -L /go/native/x86_64-unknown-linux-gnu/geos/lib/lib{geos,geos_c}.so lib
[23:05:15][Compile acceptance tests] make: *** Deleting file 'lib/libgeos_c.so'
[23:05:15][Compile acceptance tests] missing section headers
[23:05:15][Compile acceptance tests] make: *** [lib/libgeos_c.so] Error 1
[23:05:15][Compile acceptance tests] make: *** Waiting for unfinished jobs....
[23:05:15][Compile acceptance tests] patchelf --set-rpath '/usr/local/lib/cockroach/' lib/libgeos_c.so
[23:05:15][Compile acceptance tests] Makefile:755: recipe for target 'lib/libgeos_c.so' failed
[23:05:15][Compile acceptance tests] [  1%] Built target js_embed
[23:05:15][Compile acceptance tests] make: *** Deleting file 'lib/libgeos.so'
[23:05:15][Compile acceptance tests] patchelf --set-rpath '/usr/local/lib/cockroach/' lib/libgeos_c.so
[23:05:15][Compile acceptance tests] patchelf --set-soname libgeos.so lib/libgeos.so
[23:05:15][Compile acceptance tests] patchelf --replace-needed libgeos.so.3.8.1 libgeos.so lib/libgeos_c.so
[23:05:15][Compile acceptance tests] Makefile:755: recipe for target 'lib/libgeos.so' failed
[23:05:15][Compile acceptance tests] stat: No such file or directory
[23:05:15][Compile acceptance tests] make: *** [lib/libgeos.so] Error 1
[23:05:15][Compile acceptance tests] [ 47%] Built target libprotobuf
[23:05:15][Compile acceptance tests] [ 97%] Built target libprotoc
[23:05:15][Compile acceptance tests] [100%] Built target protoc
[23:05:15][Compile acceptance tests] Process exited with code 1
```

otan commented:
seems like this makes it worse, retracting...
